.github/workflows/format.yml

```diff
@@ -0,0 +1,22 @@
+name: Python Linting and Type Checks
```

Comment:
is Github workflows recommended for this project?

Reply:
The majority of the issues we ran into with the prior version of this tool would have been solved with presubmits, Github Actions are the most apparent way to do this.

Reply:
Yes, I agree and I assume we are talking about using the free tier for now. My question was more regarding if there is a formal plan on such billing and limits for this project.

Reply:
Looks like it is free for public repos and we can start with that - https://docs.github.com/en/billing/managing-billing-for-your-products/managing-billing-for-github-actions/about-billing-for-github-actions#about-billing-for-github-actions. I see other k8s projects using it as well - https://github.com/kubernetes-sigs/lws/actions. Eventually once we add more tests, we probably need an "ok-to-test" label to make sure we don't run them on every PR automatically.